modules/xmpp/strophe.util.ts

```diff
@@ -22,55 +23,48 @@ const logger = getLogger('modules/xmpp/strophe.util');
  *
  * @type {number}
  */
-let lastErrorStatus = -1;
+let lastErrorStatus: number = -1;
+
 
 /**
  * A regular expression used to catch Strophe's log message indicating that the
  * last BOSH request was successful. When there is such message seen the
  * {@link lastErrorStatus} will be set back to '-1'.
  * @type {RegExp}
  */
-const resetLastErrorStatusRegExpr = /request id \d+.\d+ got 200/;
+const resetLastErrorStatusRegExpr: RegExp = /request id \d+.\d+ got 200/;
 
 /**
  * A regular expression used to capture the current value of the BOSH request
  * error status (HTTP error code or '0' or something else).
  * @type {RegExp}
  */
-const lastErrorStatusRegExpr
+const lastErrorStatusRegExpr: RegExp
     = /request errored, status: (\d+), number of errors: \d+/;
 
 /**
  *
  */
-export default function() {
+export default function(): void {
 
-    Strophe.log = function(level, msg) {
-        // Our global handler reports uncaught errors to the stats which may
-        // interpret those as partial call failure.
-        // Strophe log entry about secondary request timeout does not mean that
-        // it's a final failure(the request will be restarted), so we lower it's
-        // level here to a warning.
+    (Strophe as IStrophe).log = function(level: number, msg: any): void {
         logger.trace('Strophe', level, msg);
         if (typeof msg === 'string'
                 && msg.indexOf('Request ') !== -1
                 && msg.indexOf('timed out (secondary), restarting') !== -1) {
             // eslint-disable-next-line no-param-reassign
-            level = Strophe.LogLevel.WARN;
+            level = (Strophe as IStrophe).LogLevel.WARN;
```

Comment:
The previous accessor should work just fine (here and elsewhere). What problem do you see?